src/dev/gethop/object_storage/s3.clj

```diff
@@ -241,6 +241,42 @@
   :args ::core/delete-object-args
   :ret  ::core/delete-object-ret)
 
+(defn- delete-objects*
+  "Delete multiple objects from S3 bucket.
+   
+   Args:
+     this - The S3 adapter containing bucket configuration
+     object-keys - Collection of object keys to delete
+   
+   Returns:
+     Map with :success? key indicating if all objects were deleted successfully
+   
+   Example:
+     (delete-objects this [\"file1.pdf\" \"file2.png\" \"file3.txt\"])"
+  [this object-keys]
+  {:pre [(and (s/valid? ::AWSS3Bucket this)
+              (coll? object-keys))]}
```

Comment:
We should use the ::core/object-keys spec defined in core (once your other PR in that repo is merge):
```suggestion
              (s/valid? ::core/object-keys object-keys))]}
```